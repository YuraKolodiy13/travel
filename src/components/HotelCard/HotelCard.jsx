import React from "react";
import './HotelCard.scss';

const HotelCard = ({item}) => {
  return (
    <div className='HotelCard'>
      <div className="HotelCard__img">
        <img src={item.photo} alt=""/>
      </div>
      <div className="HotelCard__info">
        {item.hotel.value}
        ${item.price}

        <ul>
          {item.flyInclude && <li>Включно з перельотом ({item.cityFromName})</li>}
          <li>На {item.nights} ночей</li>
          <li>Ціна за {item.adl} туриста</li>
          <li>{item.room}</li>
          <li>{item.meal.value}</li>
        </ul>
      </div>

    </div>
  )
};

export default HotelCard;