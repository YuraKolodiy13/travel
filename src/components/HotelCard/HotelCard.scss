.HotelCard{
  margin-bottom: 16px;
  box-shadow: 0 1px 6px rgba(0, 0, 0, .1);
  border-radius: 4px;
  display: flex;
  transition: .3s;
  position: relative;
  &:hover{
    box-shadow: 0 3px 12px rgba(0, 0, 0, .3);
  }
  .hotelCard__link{
    position: absolute;
    inset: 0;
  }

  .hotelCard__img{
    width: 30%;
    position: relative;
    button{
      z-index: 2;
      &:first-of-type{
        left: 0!important;
      }
      &:last-of-type{
        right: 0!important;
      }
    }
  }
  .hotelCard__info{
    width: 70%;
    padding: 12px;
    position: relative;
    z-index: 2;
    .hotelCard__header{
      display: flex;
      justify-content: space-between;
      align-items: flex-start;
      h3{
        margin: 0 0 10px;
      }
      .hotelCard__right{
        display: flex;
        align-items: center;
        span{
          color: #bfbfbf;
          border-bottom: 1px dashed;
          font-size: 12px;
          margin-right: 10px;
        }
      }
    }
    .hotelCard__time{
      cursor: pointer;
      transition: .3s;
      margin-left: 5px;
      color: #9e9e9e;
      text-decoration: underline;
      &:hover{
        text-decoration: none;
      }
    }
  }

}