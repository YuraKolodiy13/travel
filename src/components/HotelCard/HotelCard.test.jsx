import React from 'react';
import { render, screen } from '@testing-library/react';
import HotelCard from "./HotelCard";

test('renders learn react link', () => {

  const data = {
    "disablePromoCode": false,
    "HotType": 1,
    "EarlyType": 1,
    "RecommendedType": 1,
    "isHot": false,
    "isPromo": false,
    "isBestDeal": false,
    "LoadedDate": "2022-01-25T13:18:44.83+02:00",
    "operatorName": "Анекс Тур",
    "operatorIdInt": 102,
    "operatorId": "{3DF661B6-F4BB-4B95-AED1-9926925092F8}",
    "type": "plain",
    "cityFromCode": "LWO",
    "cityFromName": "Львов",
    "flyInclude": "true",
    "countryID": "25",
    "countryName": "Египет",
    "countryNameAccusativeCase": "в Египет",
    "checkIn": {"value": "2022-01-27T00:00:00+02:00", "hash": -1506999935},
    "region": {
      "RegionName": "Шарм-эль-Шейх",
      "ResortName": "Шарм-эль-Шейх",
      "RegionHash": 621,
      "ResortHash": 1486,
      "value": "Шарм-эль-Шейх Шарм-эль-Шейх",
      "hash": -2099355937
    },
    "hotel": {"hash": -1063992402, "value": "Desert View Sharm"},
    "hotelId": 109691,
    "hotelUrl": "/hotel/eg/desert-view-sharm",
    "hotelKey": "109691",
    "star": {"hash": 3, "value": "3"},
    "meal": {"hash": 1, "value": "Все включено (AI)"},
    "room": "Standard Room",
    "htplace": "DBL",
    "price": 410,
    "customPriceWd": 0,
    "customPriceWdUah": 0,
    "currency": "USD",
    "priceUAH": 12111,
    "nights": 7,
    "address": "Шарм-эль-Шейх",
    "photo": "https://img4.farvater.travel/mapkey/109691/0?size=catalog",
    "rate": "1",
    "adl": 2,
    "kids": 1,
    "ages": "1",
    "chd": 405,
    "isEarly": false,
    "SystemKey": "74m5707037035017660892c25",
    "othersAgr": null,
    "other": null,
    "isRecommended": false,
    "IsChoiceFarvater": false,
    "FarTourId": 117048032,
    "sort": 50,
    "IsFreePCRTest": false,
    "AdditionalInfo": null,
    "idsForText": {
      "roomId": 32,
      "htplaceId": 334006,
      "mealId": 1,
      "starId": 3,
      "countryid": 25,
      "longhotelId": 109691,
      "airportId": 5,
      "transferTypeId": 1,
      "touristsBits": 74,
      "reviewsCount": "0",
      "currencyId": 4
    },
    "redPriceUAH": 11964,
    "countOther": 1,
    "latitude": 27.8710159,
    "longitude": 34.3156351,
    "next": null,
    "prev": null,
    "tourFreeCancel": false,
    "isOtp": false,
    "HealthCertificate": false,
    "WorksIn2020": false,
    "promotionEndDate": null,
    "isLastSeats": false
  };

  render(<HotelCard item={data}/>);
  const list = screen.getByRole('list');
  expect(list).toBeInTheDocument();
});
