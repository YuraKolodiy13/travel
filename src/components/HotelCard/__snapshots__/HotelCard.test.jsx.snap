// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list components list snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="HotelCard"
      >
        <div
          class="hotelCard__img"
        >
          <img
            alt=""
            src="https://img4.farvater.travel/mapkey/109691/0?size=catalog"
          />
        </div>
        <div
          class="hotelCard__info"
        >
          <div
            class="hotelCard__price"
          >
            <a
              href="/countries/eg/desert-view-sharm?q=74m5707037035017660892c25"
            >
              Desert View Sharm
            </a>
            <span>
                 $
              410
            </span>
            /
            <span>
              12111
              грн
            </span>
          </div>
          <ul>
            <li>
              Включно з перельотом (
              Львов
              )
            </li>
            <li>
              Виліт на відпочинок 
              
              26
              .
              01
              <span>
                 о 
                06:05
                ,
                 
                Приліт 
                10:20
                , тривалість польоту 
                4:15
              </span>
            </li>
            <li>
              Виліт додому 
              
              21.07
              <span>
                 о 
                11:20
                ,
                 
                Приліт 
                15:40
                , тривалість польоту 
                4:20
              </span>
            </li>
            <li>
              На 
              7
               ночей
            </li>
            <li>
              Ціна за 
              3
               туриста
            </li>
            <li>
              Standard Room
            </li>
            <li>
              Все включено (AI)
            </li>
          </ul>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="HotelCard"
    >
      <div
        class="hotelCard__img"
      >
        <img
          alt=""
          src="https://img4.farvater.travel/mapkey/109691/0?size=catalog"
        />
      </div>
      <div
        class="hotelCard__info"
      >
        <div
          class="hotelCard__price"
        >
          <a
            href="/countries/eg/desert-view-sharm?q=74m5707037035017660892c25"
          >
            Desert View Sharm
          </a>
          <span>
               $
            410
          </span>
          /
          <span>
            12111
            грн
          </span>
        </div>
        <ul>
          <li>
            Включно з перельотом (
            Львов
            )
          </li>
          <li>
            Виліт на відпочинок 
            
            26
            .
            01
            <span>
               о 
              06:05
              ,
               
              Приліт 
              10:20
              , тривалість польоту 
              4:15
            </span>
          </li>
          <li>
            Виліт додому 
            
            21.07
            <span>
               о 
              11:20
              ,
               
              Приліт 
              15:40
              , тривалість польоту 
              4:20
            </span>
          </li>
          <li>
            На 
            7
             ночей
          </li>
          <li>
            Ціна за 
            3
             туриста
          </li>
          <li>
            Standard Room
          </li>
          <li>
            Все включено (AI)
          </li>
        </ul>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
